### DISCUSSION TITLE
Issue inspecting components that import TRPC

### DISCUSSION THREAD

**Discussion by hagabaka** [vuejs/devtools-v6#2042] [Q&A]
I'm working on a project using TRPC and Vue. When any component imports the TRPC client or any procedure, inspecting it leads to the error below, and nothing shows in the component tab.
![image](https://user-images.githubusercontent.com/11416/229680315-5782e693-91d2-4a1d-b572-01a99dc5e781.png)

The first line of the stacktrace points to a line inside `createTRPCProxyClient`.
```js
/**
 * @deprecated use `createTRPCProxyClient` instead
 * @internal
 */ function createTRPCClientProxy(client) {
    return createFlatProxy((key)=>{
        if (client.hasOwnProperty(key)) {
            return client[key];
        }
        return createRecursiveProxy(({ path , args  })=>{
            const pathCopy = [
                key,
                ...path
            ];
            const clientCallType = pathCopy.pop();
            const procedureType = clientCallTypeMap[clientCallType];
            const fullPath = pathCopy.join('.');
            return client[procedureType](fullPath, ...args); // <--------
        });
    });
}
```

I can avoid the error by wrapping the procedures in functions and importing only the wrappers into components. Anyone know why this is happening?

**Comment by 6XGate** [comment]
Just ran into this, and the only alternative workaround I can find is to use `defineComponent` and not expose the rpc object. I'm using `provide`/`inject` in my case and calling `inject` in a `<script setup>` triggered the issue. But, wrapping the RPC may provide a better interface though?

**Comment by Lilja** [comment]
Also ran into this. In my experience it was because my trpc client was called `client`. Renaming it to `apiClient` resolves the issue. 

I wonder if `client` is some reserved property/object that vue-devtools thinks is some internal object.

More info: https://github.com/trpc/trpc/issues/5138
